src/Toml/Parser.hs

```diff
@@ -159,10 +161,19 @@ tableNameP = lexeme $ between (char '[') (char ']') keyP
 
 -- Values
 
+decimalP :: Parser Integer
+decimalP = mkInteger <$> decimalStringP
+  where
+    decimalStringP   = fst <$> match (some digitChar >> optional (try $ many _digitsP))
+    _digitsP         = char '_' >> some digitChar
+    mkInteger        = textToInt . stripUnderscores
+    textToInt        = fst . fromRight undefined . TR.decimal
+    stripUnderscores = Text.replace "_" ""
```

Comment:
I think `Text.filter` is more efficient here than `replace`:

* https://hackage.haskell.org/package/text-1.2.3.1/docs/Data-Text.html#v:filter